src/app/collective-rewards/utils/chartUtils.ts

```diff
@@ -9,3 +19,246 @@ export const formatShort = (n: number) => {
   if (abs >= 1_000) return `${Math.round(n / 1_000)} K`
   return `${n.toLocaleString()}`
 }
+
+/**
+ * Filter data to show only the last 5 months
+ */
+const filterToLastFiveMonths = <T extends { day?: Date | number | string; start?: Date; end?: Date }>(
+  data: T[],
+  customMinDate?: number,
+): T[] => {
+  const now = Date.now()
+  const fiveMonthsAgo = now - FIVE_MONTHS_IN_MS
+  const minDate = customMinDate || fiveMonthsAgo
+
+  return data.filter(item => {
+    // For cycles
+    if (item.start && item.end) {
+      const cycleEnd = convertToTimestamp(item.end)
+      return cycleEnd >= minDate
+    }
+
+    // For backing and rewards
+    if (item.day) {
+      const itemTime = convertToTimestamp(item.day)
+      return itemTime >= minDate
+    }
```

Comment:
We could understand the data type outside the loop to do it once or even better specify the filter function outside, so you don't need to worry about it